src/hpx_wrap.cpp

```diff
@@ -61,16 +61,17 @@ namespace hpx_start
     // main entry point of the HPX runtime system
     int hpx_entry(int argc, char* argv[])
     {
-        #if defined(__linux) || defined(__linux__) || defined(linux)
+#if defined(__linux) || defined(__linux__) || defined(linux)
         // Call to the main() function
         int return_value = __real_main(argc, argv);
-        #else /* APPLE */
+#else /* APPLE */
         // call to the main() function
         int return_value = main(argc, argv);
-        #endif
+#endif
 
-        //Finalizing the HPX runtime
-        return hpx::finalize(return_value);
+        // Finalizing the HPX runtime
+        hpx::finalize();
+        return return_value;
```

Comment:
OK. I thought that finalize could return it, but if this works properly

Reply:
Yah, I thought that as well … :/
But it works this way, we verified.

Reply:
would the "proper" fix to be to make sure that `return hpx::finalize(return_value);` did what we expect it to?

Reply:
I think such an overload for `hpx::finalize()` would be possible but dangerous. We already have an overload taking a `double`.